tests/test_functional_archive_logs.py

```diff
@@ -0,0 +1,200 @@
+    """test_test_db_setup"""
+    query_result = query_test_db("SELECT 1")
+
+    assert query_result == [(1,)]
+
+
+def test_create_dummy_servers():
+    """test_create_dummy_servers"""
+    query_test_db(
+        """
+        INSERT INTO servers
+            (server_id, hostname, fqdn, ip4_address, is_enabled)
+        VALUES
+            (1, '1', '1', '192.168.0.1', 1)
+        ;
+        """
+    )
+
+    results = query_test_db(f"SELECT count(*) FROM servers WHERE is_enabled=1")[0][0]
+
+    assert results == 1
+
+
+def test_create_dummy_schedules():
+    """test_create_dummy_schedules"""
+    query_test_db(
+        """
+        INSERT INTO schedules
+            (server_id, schedule_id, interval_mask, exec_command, is_enabled, is_running)
+        VALUES
+             (1, '1-1', '* * * * *', '1-1', 1, 0)
+            ,(1, '1-2', '* * * * *', '1-2', 1, 0)
+            ,(1, '1-3', '* * * * *', '1-3', 1, 1)
+            ,(1, '1-4', '* * * * *', '1-4', 1, 0)
+        ;
+        """
+    )
+
+    results = query_test_db(f"SELECT count(*) FROM schedules WHERE is_enabled=1")[0][0]
+
+    assert results == 4
+
+
+def test_create_dummy_schedule_logs():
+    """test_create_dummy_schedule_logs"""
+    query_test_db(
+        """
+        INSERT INTO schedule_log
+            (server_id, schedule_id, full_command, start_time, end_time, returncode)
+        VALUES
+             (1, '1-1', '1-1', '2022-01-01 01:00:00', '2022-01-01 02:00:00' , 0)
+            ,(1, '1-1', '1-1', '2022-01-01 03:00:00', '2022-01-01 04:00:00' , 0)
+            ,(1, '1-1', '1-1', '2022-01-01 05:00:00', '2022-01-01 06:00:00' , 0)
+            ,(1, '1-1', '1-1', '2022-01-01 07:00:00', '2022-01-01 08:00:00' , 0)
+            ,(1, '1-1', '1-1', '2022-01-01 09:00:00', '2022-01-01 10:00:00' , 0)
+            ,(1, '1-2', '1-2', '2022-01-01 01:00:00', null , null)
+            ,(1, '1-2', '1-2', '2022-01-01 03:00:00', null , null)
+            ,(1, '1-2', '1-2', '2022-01-01 04:00:00', null , null)
+            ,(1, '1-2', '1-2', '2022-01-01 07:00:00', null , null)
+            ,(1, '1-2', '1-2', '2022-01-01 09:00:00', null , null)
+            ,(1, '1-3', '1-3', '2022-01-01 01:00:00', null , null)
+            ,(1, '1-3', '1-3', '2022-01-01 03:00:00', null , null)
+            ,(1, '1-3', '1-3', '2022-01-01 05:00:00', null , null)
+            ,(1, '1-3', '1-3', '2022-01-01 07:00:00', null , null)
+            ,(1, '1-3', '1-3', '2022-01-01 09:00:00', null , null)
+        ;
+        """
+    )
+
+    results = query_test_db(f"SELECT count(*) FROM schedule_log")[0][0]
+
+    assert results == 15
+
+
+def test_archive_schedules():
+    """test_archive_schedules"""
+    archive_schedule_log.main(1, pytest.db_test)
+
+    result = query_test_db(
+        f"SELECT (SELECT count(*) FROM schedule_log) || 'and' || (SELECT count(*) from schedule_log_historical)"
+    )[0][0]
+
+    assert result == '2and13'
+
+
+def test_archive_schedule_keep_correct_schedules():
```

Comment:
the assert in this test should imo be included in `test_archive_schedules` where the actual archiving has been done. This test is not testing a different scenario, only the result of the previous one.